src/app/model/ReportFilter.ts

```diff
@@ -12,4 +12,12 @@ export class ReportFilter {
   details?: string;
   hasCompany?: boolean;
   tags: Tag[] = [];
+
+  set hasCompanyStr(value: string) {
+    this.hasCompany = value ? value === 'true' : undefined;
```

Comment:
ou `this.hasCompany = value === 'true' || undefined`?

Reply:
Je ne pense pas car on veut distinguer le cas où `value = "false"` qui doit donner `false` du cas où `value = undefined` qui doit donner `undefined`